### PR TITLE
Implement TwoFer exercise in csharp

### PR DESCRIPTION
**Pull request by jpreese**
Just picked up the easiest unimplemented csharp exercise to get some exposure on how to implement exercises. 

The only thing I wasn't exactly sure on was the uuid generation in config.json. Looking at the configs history and previous exercise implementations, it doesn't look to be ever committed -- so I assume something is happening during the build/deployment. Is that correct?

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -1172,6 +1172,15 @@
         "Recursion",
         "Searching"
       ]
+    },
+    {
+      "slug": "two-fer",
+      "core": true,
+      "unlocked_by": "hello-world",
```
Comment (ErikSchierboom) [inline config.json]: If we use this as a core exercise, and I think we should, this (unlocked_by) should be `null`.

File: config.json [hunk 2]
```diff
@@ -1172,6 +1172,15 @@
...
+      "difficulty": 1,
+      "topics": [
+        "Strings"
```
Comment (ErikSchierboom) [inline config.json]: I would like to add the "Optional values" topic here (see other comment why)

File: config.json [hunk 3]
```diff
@@ -1172,6 +1172,15 @@
         "Recursion",
         "Searching"
       ]
+    },
+    {
```
Comment (ErikSchierboom) [inline config.json]: The ordering of the core exercises is important. Could you move this exercise's entry up, I think between the `sum-of-multiples` and `raindrops` exercises.

File: exercises/two-fer/TwoFerTest.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
+﻿// This file was auto-generated based on version 1.0.0 of the canonical data.
+
+using Xunit;
+
+namespace TwoFer
+{
+    public class TwoFerTest
```
Comment (ErikSchierboom) [inline exercises/two-fer/TwoFerTest.cs]: The test class looks good, but it does not match the [canonical-data](https://github.com/exercism/problem-specifications/blob/master/exercises/two-fer/canonical-data.json). As far as I can tell, this exercise is intended to introduce the concept of optional/default values. I think it is perfect for that, so I would like to ask you to use the same test cases as the canonical data, where the "no name given" test case should call `TwoFer.GetResponse()` (without an argument). The implementation could then assign the string parameter a default value of `null`.

**Comment by ErikSchierboom** [comment]
Good question about the Guid! @kytrinyx Is one free to chose a guid for a new exercise?

File: config.json (hunk 3 above)
Comment (kytrinyx) [inline config.json]: > The ordering of the core exercises is important.

We're actually reconsidering that. I need to open a discussion about it, but you can see the history over here: https://github.com/exercism/configlet/pull/34#issuecomment-320459371

**Comment by kytrinyx** [comment]
> it doesn't look to be ever committed

@jpreese We recently added them, and generated (and committed) them for all the existing exercises in https://github.com/exercism/csharp/pull/320

> Is one free to chose a guid for a new exercise?

Yepp. The configlet tool has a uuid command `configlet uuid` which will generate a new one, but you could use any uuid tool that is handy (e.g. https://www.uuidgenerator.net/).

I'm trying to get everything documented properly. Would you take a look at https://github.com/exercism/docs/blob/master/language-tracks/configuration/README.md and see if it makes sense?

File: config.json (hunk 3 above)
Comment (kytrinyx) [inline config.json]: Update: no we're not :) I misunderstood.

**Comment by jpreese** [comment]
@kytrinyx yeah, that makes a lot more sense. I wasn't quite sure what core was referring to and the impact it had on the exercise, same with uuid generation -- that clears it up a bit.

@ErikSchierboom I've made the requested changes sans the exercise ordering in the config file as that approach seems to still have some discussions around it.

File: exercises/two-fer/Example.cs [hunk 5]
```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace TwoFer
+{
+    public static class TwoFer
+    {
+        public static string GetResponse(string input)
```
Comment (ErikSchierboom) [inline exercises/two-fer/Example.cs]: Please also make this parameter have a default value.

**Comment by ErikSchierboom** [comment]
@jpreese Looking good! Just one small comment to fix the build.

**Comment by ErikSchierboom** [comment]
Thanks a lot @jpreese ! 🎉

**Pull request merged**